homeassistant/components/crownstone/entry_manager.py

```diff
@@ -100,6 +104,8 @@ async def async_setup(self) -> bool:
         # Makes HA aware of the Crownstone environment HA is placed in, a user can have multiple
         self.usb_sphere_id = self.config_entry.options[CONF_USB_SPHERE]
 
+        self.config_entry.runtime_data = self
+
```

Comment:
This looks like an anti-pattern to me. I think it makes more sense to assign the runtime data in `async_setup_entry`. In the end it's more versatile since you're free to change the data type itself. It's also good for consistency across different integrations.

Reply:
I think the whole "entry_manager" is an anti-pattern, not just setting the `runtime_data`

I've adjusted in 26bfae80b6813c9048ac4ba013f607c067c8d704